UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs

```diff
@@ -254,430 +118,321 @@ private static Vector3Int GetSpawnPointForOccupation(Occupation occupation)
-			ghosty.mind);
+		//Character attributes
 
-		var isAdmin = PlayerList.Instance.GetOnline(joinedViewer.connectionToClient).IsAdmin;
-		ghosty.gameObject.GetComponent<GhostSprites>().SetGhostSprite(isAdmin);
-	}
 
-	/// <summary>
-	/// Spawns an assistant dummy
-	/// </summary>
-	public static void ServerSpawnDummy(Transform spawnTransform = null)
-	{
-		if (spawnTransform == null)
+		//Character attributes
+		var Name = requestedOccupation.JobType != JobType.AI ? character.Name : character.AiName;
+		Body.name = Name;
+
+		//Character attributes
+		var playerSprites = Body.GetComponent<PlayerSprites>();
+		if (playerSprites)
 		{
-			spawnTransform = SpawnPoint.GetRandomPointForJob(JobType.ASSISTANT);
+			// This causes body parts to be made for the race, will cause death if body parts are needed and
+			// CharacterSettings is null
+			var toUseCharacterSettings = requestedOccupation.UseCharacterSettings ? character : null;
+			playerSprites.OnCharacterSettingsChange(toUseCharacterSettings);
 		}
 
-		if (spawnTransform != null)
-		{
-			var dummy = ServerCreatePlayer(spawnTransform.position.RoundToInt());
-			CharacterSheet randomSettings = CharacterSheet.GenerateRandomCharacter();
-			ServerTransferPlayer(null, dummy, null, Event.PlayerSpawned, randomSettings, null);
-
-			//fire all hooks
-			var info = SpawnInfo.Player(OccupationList.Instance.Get(JobType.ASSISTANT), randomSettings,
-				CustomNetworkManager.Instance.humanPlayerPrefab,
-				SpawnDestination.At(spawnTransform.gameObject));
-			Spawn._ServerFireClientServerSpawnHooks(SpawnResult.Single(info, dummy));
+		Body.GetComponent<DynamicItemStorage>()?.SetUpOccupation(requestedOccupation);
+		//determine where to spawn them
+		UOP.AppearAtWorldPositionServer(GetSpawnPointForOccupation(requestedOccupation));
+
+		switch (SpawnType)
+		{
+			case SpawnType.NewSpawn: //TODO Add more stuff on here
+				if (requestedOccupation != null) // && showBanner)?
+				{
+					SpawnBannerMessage.Send(
+						Body,
+						requestedOccupation.DisplayName,
+						requestedOccupation.SpawnSound.AssetAddress,
+						requestedOccupation.TextColor,
+						requestedOccupation.BackgroundColor,
+						requestedOccupation.PlaySound);
+				}
+				break;
 		}
 	}
 
-	/// <summary>
-	/// Server-side only. Creates the player object on the server side and fires server-side
-	/// spawn hooks. Doesn't transfer control to the client yet. Client side hooks should be fired after client has been
-	/// informed of the spawn
-	/// </summary>
-	/// <param name="spawnWorldPosition">world pos to spawn at</param>
-	/// <param name="occupation">occupation to spawn as</param>
-	/// <param name="characterSettings">settings to use for the character</param>
-	/// <param name="playerPrefab">prefab to spawn for the player</param>
-	/// <returns></returns>
-	private static GameObject ServerCreatePlayer(Vector3Int spawnWorldPosition, GameObject playerPrefab = null)
+	static Mind SpawnMind(CharacterSheet character)
```

Comment:
```suggestion
	private static Mind SpawnMind(CharacterSheet character)
```

always use explicit access modifiers.